app/champ/Main.hs

```diff
@@ -77,6 +87,7 @@ data ChampState = ChampState
   { champEnv          :: Exposure.Env
   , champBatchedStmts :: Seq Exposure.Stmt
   , champMultiline    :: Maybe (Seq String)
+  , champPyHandle     :: Maybe PythonHandle
```

Comment:
I don't think `champPyHandle` is a good fit for a state data type, as the value of `champPyHandle` is known from the start of the program and never changes during program execution (AFAICT). I would suggest moving this to a reader-like data type instead. This would require changing the definition of `ChampM` slightly, but I think the payoff would be worth it.